objgraph.py

```diff
@@ -168,14 +168,16 @@ def typestats(objects=None, shortnames=True):
             typename = _long_typename
         stats = {}
         for o in objects:
+            if filter and not filter(o):
+                continue
```

Comment:
This change needs a unit test.  See `TypestatsTest` in `tests.py`.  I suggest defining a new class, creating a couple of instances, then asserting that typestats()[name-of-that-class] returns 2 without a filter, and 1 with a filter that rejects one of the instances (e.g. by attribute value).